tests/rest/client/test_login_token_request.py

```diff
@@ -45,11 +45,14 @@ def prepare(self, reactor: MemoryReactor, clock: Clock, hs: HomeServer) -> None:
         self.password = "password"
 
     def test_disabled(self) -> None:
+        channel = self.make_request("POST", "/login/token", {}, access_token=None)
+        self.assertEqual(channel.code, 400)
+
         self.register_user(self.user, self.password)
         token = self.login(self.user, self.password)
 
         channel = self.make_request("POST", "/login/token", {}, access_token=token)
-        self.assertEqual(channel.code, 404)
+        self.assertEqual(channel.code, 400)
```

Comment:
Huh, do we really return 400 instead of 404? That's a surprise!

Reply:
https://github.com/matrix-org/synapse/blob/7bc110a19e6de0572b0c9513726d13298b45ced2/synapse/http/server.py#L486-L487 and https://github.com/matrix-org/synapse/blob/502f075e96b458a183952ae2be402f00b28af299/synapse/api/errors.py#L286-L292

Reply:
I think the point is that REST treats 404 as meaning "you made a well-formed request that I could interpret, but the thing you asked for doesn't exist", whereas 400 here means "you made a request that I don't recognise".

Reply:
FTR, it definitely is a 400:

`http POST https://matrix-client.matrix.org/_matrix/client/r0/login/token`

```http
HTTP/1.1 400 Bad Request
CF-Cache-Status: DYNAMIC
CF-RAY: 74ea115d6d3274b5-LHR
Connection: keep-alive
Content-Type: application/json
Date: Thu, 22 Sep 2022 09:38:28 GMT
Server: cloudflare
Transfer-Encoding: chunked
access-control-allow-headers: X-Requested-With, Content-Type, Authorization, Date
access-control-allow-methods: GET, HEAD, POST, PUT, DELETE, OPTIONS
access-control-allow-origin: *
cache-control: no-cache, no-store, must-revalidate
permissions-policy: interest-cohort=()
synapse-trace-id: 698ffcd5594e1d56

{
    "errcode": "M_UNRECOGNIZED",
    "error": "Unrecognized request"
}
```